roadmap/implementers-guide/src/runtime/scheduler.md

```diff
@@ -1,3 +1,227 @@
+cores are allowed (winning two cores in a single block). Likely a `max_parallel`
+parameter. Then the user can submit a single extrinsic, with only a single
+signature biding for e.g. 10 cores. Internally we treat those as indidivual
+equal bids (all the same price), competing individually with all the other bids.
+Therefore it is possible that only a fraction of the desired cores will actually
+be won. E.g. just one or two out of the ten desired.
+
+
+## Where is the money going?
+
+TODO: Bids and fees - where are they spent, what pallet do we need to interact here
+with?
+
+## Scheduler/Collators/Validators
+
+1. Collators need to know in advance when they should connect to which validator
+   group to prepare and send a collation.
+2. Collators need to know in advance what collators for which
+   parachains/parathreads they should accept connections and collations from.
+3. Users need to know whether they won a slot or not - seeing assigned cores
+   could be one way of achieving that goal. (Likely not very suitable though).
+
+### Interface to the scheduler
+
+Once we found claims for cores, we need to get them to the scheduler. This
+interface will roughly be:
+
+Give me claims for this claim queue size for this number of cores. For
+simplicity we make the number of cores per assignment provider a configuration
+that can only be changed on session boundaries.
+
+Basically the scheduler needs to be able to provide core assignments for the
+look ahead period required by asynchronous backing. The assignment providers
+(like the parathread one above) need to be able to provide those, given a
+configured number of available cores to the provider.
+
+Assuming we treat cores separately and sequentially (see questions below), an
+assignment provider could return an enum like this:
+
+```rust
+enum Assignment {
+  // Parathreads
+  List(Vec<Claim>),
+  // slot sharing/slow parachains:
+  // Useful for parachains, which want a 12 second block time. - With async
+  backing, all parachains will be of this type in the beginning:
+  Alternating2(Claim, Claim),
+  // 18 second parachains ... cost a third.
+  Alternating3(Claim, Claim, Claim),
+  // For parachains/booster
+  Single(Claim),
+}
+```
+
+per core.
+
+So the interface could look something like this:
+
+```rust
+trait AssignmentProvider {
+  /// Upcoming assignments that are ready from the provider.
+  pub fn lookahead(&self) -> UpcomingAssignments
+  /// Tell provider that an item in the claim queue at the given index has been
+  served (or timed out) - in any case core became free.
+  pub fn pop_assignment(index: AssignmentIndex);
+}
+
+
+```
+
+^^ To initiate a new round `set_num_cores`... needs to be called with a
+`RoundId`, can only be called once per round.
+
+Questions:
+
+Can different cores in a block and cores across blocks be treated the same? E.g.
+if a parathread a parachain has two claims, would it be legal to give it two
+cores in the same block?
```

Comment:
I'm not following. What's the scenario here?

Reply:
I think that is already obsolete. I was referencing something you wrote, but I think I misunderstood.